VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/hearing-controls/hearing-controls.component.ts

```diff
@@ -124,35 +125,37 @@ export class HearingControlsComponent implements OnInit, OnDestroy {
         }
     }
 
-    handleHearingCountdownComplete(conferenceId: string) {
+    async handleHearingCountdownComplete(conferenceId: string) {
         if (this.isJudge && conferenceId === this.conferenceId) {
-            this.resetMute();
+            await this.resetMute();
         }
 
         if (!this.isJudge && conferenceId === this.conferenceId && !this.audioMuted) {
             this.logger.info(`${this.loggerPrefix} Countdown complete, muting participant`, this.logPayload);
-            this.toggleMute();
+            await this.toggleMute();
         }
+        await this.publishMediaDeviceStatus();
```

Comment:
not needed?

Reply:
When the countdown is complete, the remote mute is removed but the local mute is still on. Without this end of countdown publish, the judge would still see the participant as not muted although they actually are.

I ran the scenario a few times and the order of the messages never arrived before the judge participant list was set up and thus it always missed the local mute message.  I added this bit in to ensure the judge would have the correct mute statuses after the countdown was complete.

Reply:
hmm sounds like it might still be a race condition if the judge has a slow connection? maybe we do need a way for the judge to request all participants to send their status?

Reply:
talked through this and only publish the result if the mute is not toggled